docs/api-reference/web/adapters/metamask.mdx

```diff
@@ -1,65 +1,128 @@
+
+## [`@web3auth/metamask-adapter`](https://npmjs.com/package/@web3auth/metamask-adapter)
+
+Metamask Adapter allows user to login with metamask wallet. You can read more about metamask wallet [here](https://docs.metamask.io/guide/).
+
+## Basic Details
+
+---
 
 #### Adapter Name: `metamask`
 
-#### Package Name: `@web3auth/metamask-adapter`
+#### Package Name: [`@web3auth/metamask-adapter`](https://npmjs.com/package/@web3auth/metamask-adapter)
 
 #### authMode: `DAPP`
 
 #### chainNamespace: `EIP155`
 
 #### Default: `YES`
 
-#### Description:
+## Installation
 
-Metamask Adapter allows user to login with metamask wallet. You can read more about metamask wallet [here](https://docs.metamask.io/guide/).
+---
+
+<Tabs
+  defaultValue="npm"
+  values={[
+    { label: "npm", value: "npm" },
+    { label: "yarn", value: "yarn" },
+  ]}
+>
+
+<TabItem value="npm">
+
+```shell
+npm install --save @web3auth/metamask-adapter
+```
+
+</TabItem>
+
+<TabItem value="yarn">
+
+```shell
+yarn add @web3auth/metamask-adapter
+```
+
+</TabItem>
 
-### Interface
+</Tabs>
+
+## Arguments
 
 ---
 
+<Tabs
+  defaultValue="table"
+  values={[
+    { label: "Table", value: "table" },
+    { label: "Interface", value: "interface" },
+  ]}
+>
+
+<TabItem value="table">
+
+| Parameter     | type                |
+| ------------- | ------------------- |
+| `chainConfig` | `CustomChainConfig` |
+
+</TabItem>
+
+<TabItem value="interface">
+
 ```ts
 interface MetamaskAdapterOptions {
   chainConfig?: CustomChainConfig;
 }
 ```
 
-### Params
+</TabItem>
 
----
+</Tabs>
+
+### Custom Chain Config
+
+#### `chainConfig`
+
+:::caution
+
+While you can pass your `chainConfig`, it is not required since you can directly pass it over to the `Web3Auth`/ `Web3AuthCore` configuration while
+instantiating it.
+
+Read more about it in their respective sections:
+
+- [`web3auth/web3auth`](./web3auth/initialize#adding-a-custom-chain-configuration)
+- [`web3auth/core`](./core/initialize#adding-a-custom-chain-configuration)
 
-| Parameter   | type                                       |
-| ----------- | ------------------------------------------ |
-| chainConfig | [CustomChainConfig](#common_configuration) |
+:::
 
-### Example
+## Example
 
 ---
 
 ```ts
-   import `Web3Auth` class from  `@web3auth/web3auth`;
-   import { MetamaskAdapter } from '@web3auth/metamask-adapter'
-   const web3auth = new Web3Auth(options);
-   const adapter = new MetamaskAdapter({
-        chainNamespace: CHAIN_NAMESPACES.EIP155,
-        chainId: "0x3",
-        rpcTarget: "https://ropsten.infura.io/v3/776218ac4734478c90191dde8cae483c",
-        displayName: "ropsten",
-        blockExplorer: "https://ropsten.etherscan.io/",
-        ticker: "ETH",
-        tickerName: "Ethereum",
-      })
-
-  // it will add/update  the metamask adapter in to web3auth class
-  web3auth.configureAdapter(adapter);
+import { MetamaskAdapter } from "@web3auth/metamask-adapter";
+const metamaskAdapter = new MetamaskAdapter({
```

Comment:
thats incorrect. Params are incorrectly passed here, should be inside chainConfig.

Reply:
removed the params, not needed